### ISSUE TITLE
[BUG] Velocity, unable to use a number as string for key

### ISSUE THREAD

**Issue by RunnenLate** [nextgenhealthcare/connect#6118]
**Describe the bug**
Trying to utilize velocity to code template to put a mapped variable with nested object you can't utilize a key value that is set as a strictly number cast as a string.  So if the key is 0,1,2,3... if you try to call that it will not work.  see the screen shots below 

**To Reproduce**
Setting the values:
![image](https://github.com/nextgenhealthcare/connect/assets/2079191/9f9e85cc-1fb5-4bbf-ade4-f39513ccd7af)
When is being sent:
![image](https://github.com/nextgenhealthcare/connect/assets/2079191/c85b78e9-3265-4522-ac43-7184329cd88f)
Received: 
![image](https://github.com/nextgenhealthcare/connect/assets/2079191/26f16ef7-704f-44e7-9833-c61230e0255d)

**Expected behavior**
Find value based on key values provided

**Actual behavior**
Sending the velocity code not finding the keys to grab the value.   I've tried a combinations of different things. 

${value.1.keyFromStringNumber}

${value.get("1").keyFromStringNumber}

#set($String = "1")
${value.get($String).keyFromStringNumber}

**Environment (please complete the following information):**
 - OS: [e.g. Linux (CentOS), Windows 2019]
 - Java Distribution/Version [e.g. OpenJDK 17, Java 8 (201)]
 - Connect Version [e.g. 4.4.1]

**Workaround(s)**
Don't use number as a key value

**Comment by RunnenLate** [comment]
Maybe this isn't a bug with mirth but with Rhino, either way not great with velocity.  Maybe someone knows how to do this.  At the very least it is visibility to other people who might run into the same issue.

**Comment by cmcquillen** [comment]
tldr; If the key is a valid integer, you pass it as an integer without quotes. In your example above, it would be:
`${value.get(1).keyFromStringNumber}`

Building my object:
![image](https://github.com/user-attachments/assets/68d21f4e-676e-4a41-8adf-e36458c1329b)

Template:
```
#foreach($key in $myobject.keySet())
"$key" is $key.class.name
#end

${myobject.get('a')}
${myobject.get("a")}

${myobject.get(00512)}
${myobject.get('00512')}
${myobject.get("00512")}

${myobject.get(20.5)}
${myobject.get('20.5')}
${myobject.get("20.5")}

${myobject.get(0.05)}
${myobject.get('0.05')}
${myobject.get("0.05")}

${myobject.get(100)}
${myobject.get('100')}
${myobject.get("100")}

${myobject.get(101)}
${myobject.get('101')}
${myobject.get("101")}

${myobject.get(102)}
${myobject.get('102')}
${myobject.get("102")}

${myobject.get(103)}
${myobject.get('103')}
${myobject.get("103")}
```

Output:
```
"a" is java.lang.String
"00512" is java.lang.String
"20.5" is java.lang.String
"0.05" is java.lang.String
"100" is java.lang.Integer
"101" is java.lang.Integer
"102" is java.lang.Integer
"103" is java.lang.Integer

string_a_as_key
string_a_as_key

${myobject.get(00512)}
number_00512_as_key
number_00512_as_key

${myobject.get(20.5)}
number_20.5_as_key
number_20.5_as_key

${myobject.get(0.05)}
number_0.05_as_key
number_0.05_as_key

number_100_as_key
${myobject.get('100')}
${myobject.get("100")}

java_Integer_101_as_key
${myobject.get('101')}
${myobject.get("101")}

java_integer_102_as_key
${myobject.get('102')}
${myobject.get("102")}

java_string_103_as_key
${myobject.get('103')}
${myobject.get("103")}
```

Non-integer numbers are treated as strings, and need quotes. Integers, regardless of the Class used to create them, are parsed out of the object as java.lang.Integer.

**Comment by RunnenLate** [comment]
Interesting, my initial hope for this was to create a globalMap of username and passwords that's populated from a table in the database and pull them via the channelId and destinationId

Have something like this: 
```
var mapping= {
	"341234-2134dd23-d32d3-23dd32-2d2-d334": {
		"0": {
			"username": "USERNAME",
			"password": "PASSWORD"
		}
	},
channelMap.put('mapping', mapping)
```

Then just use the same Velocity value in every channel 
```
${mapping[${channelId}].[${destinationId}].username} 
${mapping[${channelId}].[${destinationId}].password} 
```
But destinationId is not a thing so that won't work anyways.  I just ended up defining them as d1, d2, d3 .... in the database and just have to remember to change the value based on what the destination number is. `${mapping[${channelId}].d1.password} `

**Comment by cmcquillen** [comment]
If you want the destination metadata ID, it is available.
Transformer in each destination (be sure to use connectorMap):
``` js
connectorMap.put("connectorId", String(connectorMessage.getMetaDataId()));
```

Template:
`${connectorId}`

Output is the metadata Id of that destination like:
`1`

**Comment by RunnenLate** [comment]
The goal was to have it all within the velocity variable so if it could be done somehow like this `${mapping[${channelId}][${connectorMessage.getMetaDataId()}].password} ` 

I suppose I could always add that code for the connector map code to my deployment script so it's added to each destination automatically, but for right now the juice isn't worth the squeeze.  (When I say deployment script I mean a python script I wrote to automatically push between environments via the api and git, not the Deploy script within Mirth)

I do appreciate the clarification on the int/string classification .

**Comment by cmcquillen** [comment]
I don't know if it's documented anywhere, but this velocity code gets the metadata ID of the current destination:
`${message.metaDataId}`

It appears to be equivalent to calling `connectorMessage.getMetaDataId()` in a channel script as `${message}` has Class of:
`com.mirth.connect.userutil.ImmutableConnectorMessage`
